### DISCUSSION TITLE
`x-on:click` possible vulnerability.

### DISCUSSION THREAD

**Discussion by asmeeee** [alpinejs/alpine#1140] [1. Help] [answered]
Hi! Do you see any security issue telling an element to call a global function like so:

```javascript
window.myFunction = () => doSomething();

<button x-on:click="myFunction" />
```

And then for the user to override this function in the console to manipulate the state and outcome in case of not properly taking care for the logic following the input and output of that function.

Is it better to work with `x-data` scope and provide the necessary `window.myFunction` only for current scope, since (as I understand) you would not be able to override whatever was initially scoped from `x-data` therefore the only secure way to work with Alpine is to always declare what you need inside `x-data` and for the child elements to use only what is declared in this scope ignoring global `window.` properties/functions?

**Comment by SimoTod** [comment, marked as answer]
You can override a function in x-data if you really want to so from a security point of view, there's no much different.

Javascript runs locally, from the console a user can only hack himself which is a bit silly. The security threat comes from third party scripts but that's generally true. If you include in your page a malicious script, that script can do nasty things even without alpine. 

If you are concerned about data validation, that always needs to happen server side (regardless you using alpine).

Are you worried about a specific scenario?

**Reply by asmeeee** [reply]
Thank you for your reply! I do not have a specific scenario on my mind, but I am concerned on what possible mistakes developers could make and how to prevent them.
